src/tools/kotlin-codegen-shared.ts

```diff
@@ -31,8 +31,10 @@ export function generateConnectionSpecType(connection: HandleConnectionSpec, nod
   }
 
   return (function generateType(type: Type): string {
-    if (type.isEntity) {
-      const node = nodes.find(n => n.schema.equals(type.getEntitySchema()));
+    if (type.isEntity || type.isVariable) {
+      const node = type.isEntity
+        ? nodes.find(n => n.schema && n.schema.equals(type.getEntitySchema()))
```

Comment:
schema is no longer nullable, so you can remove this branch, but you should check that n.variableName is null

Reply:
Good catches, thanks.